flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js

```diff
@@ -1427,9 +1427,12 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
      * Attempts to connect to a local websocket server
      */
     connectWebsocket(port) {
+
         if (this.socket) {
-            this.socket.close();
-            this.socket = undefined;
+            if (this.socket.readyState) {
```

Comment:
What about readyState == 2 and == 3? Do we want to call socket.close when it's already `CLOSING` or `CLOSED`?

Reply:
This is already the case:
```javascript
if (this.socket) {
    if (this.socket.readyState) { 
        console.log('Attempting to close socket: readyState=' + this.socket.readyState);
        this.socket.close();
    }
    delete this.socket;
}
```
All non 0 will be closed